src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
```

Comment:
@KjellKod no you are not wrong, as part of my home library building I went through the allocation/deallocation code. void *s allocated in one process cannot be sent and accessed in another process.